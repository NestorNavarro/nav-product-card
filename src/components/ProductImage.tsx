import React, { useContext } from "react";

import noImg from "../assets/no-image.jpg";
import { ProductContext } from "./ProductCard";

import stylesCss from "../styles/styles.module.css";

export interface ProductImageProps {
    img ?: string;
    title ?: string;
    className ?: string;
    style ?: React.CSSProperties;
}

export const ProductImage = ({ img = "", title = "", className = "", style } : ProductImageProps) => {
    const { product } = useContext(ProductContext);
    let imgToShow: string;

    if ( img ) {
        imgToShow = img;
    } else if ( product.img ) {
        imgToShow = product.img
    } else {
        imgToShow = noImg;
    }

    return (
        <img 
            style={style}
            className={`${stylesCss.productImg} ${className}`}
            src={imgToShow ? imgToShow : noImg}
            alt={title}
        />
    )
}

export default ProductImage