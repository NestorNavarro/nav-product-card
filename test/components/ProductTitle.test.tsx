import React from "react";
import renderer from "react-test-renderer";
import { ProductTitle, ProductCard } from "../../src/components";
import { product1 } from "../data/product";

describe('ProductTitle', () => { 
    test('should to show a title', () => { 
        const wrapper = renderer.create(
            <ProductTitle title="Custom Title" className="custom-class" />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should to show name product into the component', () => { 
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
