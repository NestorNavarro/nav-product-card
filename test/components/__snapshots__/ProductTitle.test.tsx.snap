// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should to show a title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Title
</span>
`;

exports[`ProductTitle should to show name product into the component 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - without img
  </span>
</div>
`;
